File: src/main/java/com/feiliks/common/PasswordUtil.java
package com.feiliks.common;

import org.apache.commons.codec.digest.DigestUtils;

public class PasswordUtil {

    public static String hash(String username, String plainPassword) {
        return DigestUtils.sha256Hex(
                DigestUtils.sha256Hex(String.format(
                        "%s%s", username, plainPassword)));
    }

}

File: src/main/kotlin/com/feiliks/common/PasswordUtil.kt
package com.feiliks.common

import org.apache.commons.codec.digest.DigestUtils

object PasswordUtil {

    fun hash(username: String, plainPassword: String): String {
        return DigestUtils.sha256Hex(
                DigestUtils.sha256Hex(username + plainPassword))
    }

}
